src/main/java/edu/kit/kastel/sdq/artemis4j/client/ArtemisClient.java

```diff
@@ -0,0 +1,181 @@
+/* Licensed under EPL-2.0 2024. */
+package edu.kit.kastel.sdq.artemis4j.client;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.StreamReadFeature;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+import edu.kit.kastel.sdq.artemis4j.ArtemisNetworkException;
+import okhttp3.Cookie;
+import okhttp3.CookieJar;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ArtemisClient {
+    public static final ObjectMapper MAPPER = createObjectMapper();
+
+    private static final Logger log = LoggerFactory.getLogger(ArtemisClient.class);
+    private static final String COOKIE_NAME_JWT = "jwt";
+    private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+
+    private final ArtemisInstance artemis;
+    private final String jwtToken;
+    private final String password;
+    private final OkHttpClient client;
+
+    public static ArtemisClient fromUsernamePassword(ArtemisInstance artemis, String username, String password) throws ArtemisNetworkException {
+        if (username == null || password == null) {
+            throw new IllegalArgumentException("Username and password must not be null");
+        }
+
+        // The client is just used for login
+        var client = new OkHttpClient();
+
+        var payload = ArtemisClient.encodeJSON(new AuthenticationDTO(username, password));
+        var request = new Request.Builder().url(artemis.url(List.of("public", "authenticate"), null)).post(payload).build();
+
+        String jwtToken;
+        try (var response = client.newCall(request).execute()) {
+            throwIfStatusUnsuccessful(response);
+
+            // jwt=JWT_CONTENT_HERE; Path=/; Max-Age=2592000; Expires=Sun, 26 Feb 2023
```

Comment:
what are these comments for

Reply:
They show what the cookie's content looks like. I'm going to add an explaining comment